src/frequenz/sdk/actor/_runner.py

```diff
@@ -0,0 +1,27 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Utility functions to run and synchronize the execution of actors."""
+
+
+import asyncio
+from typing import Any
+
+from ._decorator import BaseActor
+
+
+async def run(*actors: Any) -> None:
```

Comment:
Did you try to use `BaseActor` instead of `Any`?
async def run(*actors: `BaseActor`) -> None:

Reply:
yes, I did try it and I'm afraid it didn't work. This is due to the way the actor decorator instantiate an internal class which is the `BaseActor`. This is why I've done the type check at runtime and I added an inline comment to highlight the reason of the type check at runtime